STACpopulator/implementations/CMIP6_UofT/add_CMIP6.py

```diff
@@ -130,8 +132,8 @@ def main(*args: str) -> int:
     parser = argparse.ArgumentParser()
     add_parser_args(parser)
     ns = parser.parse_args(args or None)
-    return runner(ns)
-
+    ns.populator = "CMIP6_UofT"
+    return cli.run(ns)
```

Comment:
I think this is potentially dangerous if we add other sub-`ArgumentParser` commands than the top-level CLI `run` command which might not have the `populator` attribute. 

Since the `run` function (not to confuse with the `run` subparser command) only does `implementation_modules()[ns.populator].runner(ns)`, there's basically no difference from doing `runner(ns)` (as before), and there's no need to inject the `populator` argument anymore.

Reply:
This is necessary so that the logging options are applied when this file is executed directly from the command line.

> I think this is potentially dangerous if we add other sub-ArgumentParser commands than the top-level CLI run command which might not have the populator attribute.

Agreed, if that happens, then we'd have to modify this code as well. The simpler and more maintainable thing to do would be to remove the option to call populator scripts directly from the command line.

Reply:
I'm not sure to understand why the logging options cause this? `populator` is only defined by the `run` subparser. What causes `populator` to become an issue if calling the script directly since `run` would be bypassed entirely?

Reply:
I gave it a try, and no error on my end (beside the unrelated schema validation of course). 

![{BE7604F5-C019-4211-A68C-3435813DA442}](https://github.com/user-attachments/assets/ae7970a5-8091-42e6-8102-2b26a4697885)

Reply:
> I'm not sure to understand why the logging options cause this?

There's no error but the logging options are applied here https://github.com/crim-ca/stac-populator/blob/b353febe2d703a49f9574cbee9feb3feb131108f/STACpopulator/cli.py#L49

So if you don't run it through the CLI functions then the logging options are never applied. We could have added a call to `setup_logging` in every `runner` function as well but that seemed like we're adding boilerplate (which we're actively trying to remove in #64)

Reply:
OK. But can we at least resolve the `populator` name automatically rather than a hard coded value? I really dislike having different name resolution strategies by different part of the code. If the directory name is the "reference" for the populator name, then everywhere should resolve it the same way.